src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua

```diff
@@ -149,10 +158,32 @@ local function tabulateCommon( bPaged, ... )
 end
 
 function tabulate( ... )
+    tArgs = { ... }
+    for i=1,#tArgs,2 do
```

Comment:
Instead of wrapping the varargs in the table and indexing it, you can use [`select`](https://www.lua.org/pil/5.2.html) to get the length and index it.

For instance, consider `tabulate("foo", "bar", nil, nil)` - this won't produce an error, as the length of the table is only 2. Using `select('#', ...) will return 4 - and so you'll error as expected. [`print` does something similar](https://github.com/dan200/ComputerCraft/blob/master/src/main/resources/assets/computercraft/lua/bios.lua#L262-L269).